core/src/main/java/org/elasticsearch/index/codec/CodecService.java

```diff
@@ -83,17 +63,40 @@ public MapperService mapperService() {
     }
 
     public Codec codec(String name) {
-        Codec codec = codecs.get(name);
-        if (codec == null) {
+        switch (name) {
+        case DEFAULT_CODEC:
+            if (mapperService == null) {
+                return new Lucene53Codec();
+            }
+            return new PerFieldMappingPostingFormatCodec(Mode.BEST_SPEED, mapperService, logger);
+        case BEST_COMPRESSION_CODEC:
+            if (mapperService == null) {
+                return new Lucene53Codec(Mode.BEST_COMPRESSION);
+            }
+            return new PerFieldMappingPostingFormatCodec(Mode.BEST_COMPRESSION, mapperService, logger);
+        case LUCENE_DEFAULT_CODEC:
+            return Codec.getDefault();
+        default:
+            if (Codec.availableCodecs().contains(name)) {
+                return Codec.forName(name);
+            }
             throw new IllegalArgumentException("failed to find codec [" + name + "]");
```

Comment:
Yeah - but the message is different and I didn't want to change it. I'm fine if you want.